apps/nerves_hub_www/lib/nerves_hub_www_web/controllers/org_certificate_controller.ex

```diff
@@ -52,24 +61,38 @@ defmodule NervesHubWWWWeb.OrgCertificateController do
     end
   end
 
+  require IEx
+
   def create(
-        %{assigns: %{org: org}} = conn,
-        %{"ca_certificate" => %{"cert" => %{path: upload_path}}} = params
+        %{assigns: %{org: org, user: user}} = conn,
+        %{
+          "ca_certificate" => %{
+            "cert" => %{path: cert_upload_path},
+            "csr" => %{path: csr_upload_path}
+          }
+        } = params
       ) do
-    with {:ok, cert_pem} <- File.read(upload_path),
+    with {:ok, cert_pem} <- File.read(cert_upload_path),
+         {:ok, csr_pem} <- File.read(csr_upload_path),
          {:ok, cert} <- X509.Certificate.from_pem(cert_pem),
+         {:ok, csr} <- Certificate.from_pem(csr_pem),
+         :ok <- validate_csr(conn, cert, csr),
          serial <- Certificate.get_serial_number(cert),
          aki <- Certificate.get_aki(cert),
          ski <- Certificate.get_ski(cert),
-         {not_before, not_after} <- Certificate.get_validity(cert),
+         {cert_not_before, cert_not_after} = cert_validity <- Certificate.get_validity(cert),
+         {_csr_not_before, _csr_not_after} = csr_validity <- Certificate.get_validity(csr),
+         :ok <- check_validity(cert_validity),
+         :ok <- check_validity(csr_validity),
          params <- %{
            serial: serial,
            aki: aki,
            ski: ski,
-           not_before: not_before,
-           not_after: not_after,
+           not_before: cert_not_before,
+           not_after: cert_not_after,
            der: X509.Certificate.to_der(cert),
-           description: Map.get(params["ca_certificate"], "description")
+           description: Map.get(params["ca_certificate"], "description"),
+           jitp: Map.get(params, "jitp")
```

Comment:
```suggestion
           jitp: Map.get(params["ca_certificate"], "jitp")
```

`jitp` is nested inside of `ca_certificate` fields